lib/codecs/src/decoding/format/gelf.rs

```diff
@@ -0,0 +1,243 @@
+        let mut secs = 0;
+        let mut nsecs = 0;
+        if val.is_f64() {
+            secs = f64::trunc(val.as_f64().unwrap()) as i64;
+            nsecs = f64::fract(val.as_f64().unwrap()) as u32;
+        } else if val.is_i64() {
+            secs = val.as_i64().unwrap();
+        } else if val.is_u64() {
+            secs = val.as_u64().unwrap() as i64;
+        }
+        let naive = NaiveDateTime::from_timestamp(secs, nsecs);
+        DateTime::<Utc>::from_utc(naive, Utc)
+    }
+
+    /// Attemps to parse an integer from a Value.
+    fn parse_number(&self, val: &Value) -> Result<i64, vector_core::Error> {
+        if val.is_number() {
+            let mut number = 0;
+            if val.is_f64() {
+                number = f64::round(val.as_f64().unwrap()) as i64;
+            } else if val.is_u64() {
+                number = val.as_u64().unwrap() as i64;
+            } else if val.is_i64() {
+                number = val.as_i64().unwrap();
+            }
+            Ok(number)
+        } else if val.is_string() {
+            match val.as_str().unwrap().parse::<i64>() {
+                Ok(number) => Ok(number),
+                Err(_) => match val.as_str().unwrap().parse::<f64>() {
+                    Ok(number) => Ok(f64::round(number) as i64),
+                    Err(_) => Err(format!(
+                        "Event field {} does not match GELF spec version {}: must be a number",
+                        VERSION, GELF_VERSION
+                    )
+                    .into()),
+                },
+            }
+        } else {
+            Err(format!(
+                "Event field {} does not match GELF spec version {}: must be a number",
+                VERSION, GELF_VERSION
+            )
+            .into())
+        }
+    }
+
+    /// Builds a LogEvent from the parsed GelfMessage.
+    /// The logic does not follow strictly the documented GELF standard, it more closely
+    /// follows the behavior of graylog itself, which is more relaxed.
+    fn message_to_event(&self, parsed: &GelfMessage) -> vector_core::Result<Event> {
+        if parsed.message.is_none() && parsed.short_message.is_none() {
+            return Err("Event must contain the field 'short_message'".into());
+        }
+
+        let message = if let Some(msg) = &parsed.short_message {
+            msg
+        } else {
+            parsed.message.as_ref().unwrap()
+        };
+        let mut event = Event::from(message.to_string());
+        let log = event.as_mut_log();
+
+        let timestamp_key = log_schema().timestamp_key();
+
+        // GELF spec defines the version as 1.1 which has not changed since 2013
+        // but graylog server does not reject any event which does not specify a version
+
+        if let Some(host) = &parsed.host {
+            log.insert(HOST, host.to_string());
+        }
+        // TODO graylog sets field 'host' to IP address if not specified. I'm not seeing a clear way to get the IP...
```

Comment:
Good question and why I left the TODO here :) I am happy to look in to resolving it in this PR but wasn't sure if it would turn into a lot of changes. I can inspect the upstream code from `parse()` and see if I can get a sense for it.

Reply:
That would likely depend on the source itself (assuming IP here is the client IP), some source will enrich with that metadata. We could look for where we'd expect it (from those sources) but I don't know how we'd handle it otherwise.

Reply:
As it sounds like it isn't entirely straightforward, perhaps a new issue to investigate this would be a good idea?

Reply:
Opened : https://github.com/vectordotdev/vector/issues/13323
, I'll reference in the TODO